vm/src/import.rs

```diff
@@ -144,3 +144,35 @@ fn find_source(vm: &VirtualMachine, current_path: PathBuf, name: &str) -> Result
         None => Err(format!("No module named '{}'", name)),
     }
 }
+
+pub fn remove_importlib_frames(vm: &VirtualMachine, exc: &PyObjectRef) -> PyObjectRef {
+    let always_trim = objtype::isinstance(exc, &vm.ctx.exceptions.import_error);
+
+    if let Ok(tb) = vm.get_attribute(exc.clone(), "__traceback__") {
+        if objtype::isinstance(&tb, &vm.ctx.list_type()) {
+            let mut tb_entries = objsequence::get_elements_list(&tb).to_vec();
+            tb_entries.reverse();
+            let mut new_tb = Vec::with_capacity(tb_entries.len());
```

Comment:
Since this function is basically a filtering of a list, this might be an excellent opportunity to use the `filter_map` function, or `filter` function on rust iterators, and then use `collect` on the filtered iterator.

Something along this idea:

```rust
let new_tb = tb_entries.iter()
    .filter(|tb_entry| {
       // fetch filename etc..
       // funky filtering here, if file_name != frozen etc...
    })
    .collect();


```

Reply:
This is only a simple filter if the error is from outside the `importlib` code. That is why we remove only chunks that end in `_call_with_frames_removed`.